tests/validation/validation_data.cpp

```diff
@@ -15,6 +16,38 @@ namespace mc {
 
 trace_io g_trace_io;
 
+util::path trace_io::find_datadir() {
+    // If environment variable is set, use that in preference.
+    if (const char* env_path = std::getenv("NMC_DATADIR")) {
+        return util::path(env_path);
```

Comment:
It isn't clear whether the path returned from this function is meant to be valid or not. The later tests test if paths exist.